cnd/src/db/mod.rs

```diff
@@ -28,66 +32,116 @@ impl Sqlite {
     /// When this returns an Sqlite database exists at 'db', a
     /// successful connection to the database has been made, and
     /// the database migrations have been run.
-    pub fn new(db: Option<PathBuf>) -> anyhow::Result<Sqlite> {
-        let file = db
-            .or_else(default_db_path)
-            .context("failed to determine default path for database ")?;
-        let db = Sqlite { db: file };
+    pub fn new(location: Location<'_>) -> anyhow::Result<Self> {
+        let db = match location {
+            Location::OnDisk(path) => {
+                if path == Path::new(":memory:") {
+                    anyhow::bail!("use Location::InMemory if you want an in-memory database!")
+                }
+
+                ensure_folder_tree_exists(path)?;
```

Comment:
```suggestion
                ensure_directory_tree_exists(path)?;
```
It's bad enough only half the team are Linux hackers without using OSX terminology too :)

Reply:
Today I learned something.